crates/matrix-sdk-base/src/store/memory_store.rs

```diff
@@ -577,7 +630,14 @@ impl StateStore for MemoryStore {
     }
 
     async fn get_presence_event(&self, user_id: &UserId) -> Result<Option<Raw<PresenceEvent>>> {
-        self.get_presence_event(user_id).await
+        Ok(self.get_presence_events(iter::once(user_id)).await?.into_iter().next())
+    }
+
+    async fn get_presence_events(
+        &self,
+        user_ids: &[OwnedUserId],
+    ) -> Result<Vec<Raw<PresenceEvent>>> {
+        self.get_presence_events(user_ids.iter().map(|u| u.as_ref())).await
```

Comment:
I've a preference for:

```suggestion
        self.get_presence_events(user_ids.iter().map(AsRef::as_ref)).await
```

but it's not mandatory.